File: web/src/models/blog.ts
export type BlogSummary = {
  id: string;
  title: string;
  description: string;
  tags: string[] | undefined;
  createdAt: string;
};

export type Blog = {
  id: string;
  userId: number;
  title: string;
  description: string;
  content: string;
  comments: number;
  tags: string[] | undefined;
  createdAt: string;
  updatedAt: string;
};

export type BlogRequest = {
  title: string;
  description: string;
  tags: string[] | undefined;
  content: string;
};

File: server/src/database/entities/blog.rs
use chrono::{DateTime, Utc};
use serde::Serialize;
use uuid::Uuid;

#[derive(Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Blog {
    pub id: Uuid,
    pub title: String,
    pub description: String,
    pub content: String,
    pub comments: i64,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tags: Option<Vec<String>>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
